src/test/java/com/springboot/greencommute/controllerTest/JobControllerTest.java

```diff
@@ -67,4 +74,49 @@ void getJobByIdTest() throws Exception {
         verify(jobService).getJobById(1);
         verify(jobService,times(1)).getJobById(1);
     }
+
+    @Test
+    void getAllJobTestByFiltering() throws Exception {
+     Job job = new Job(1,"dev","Koch",null,null);
+        Skill skill = new Skill(1,"singing",null);
+        List<Skill> skillList = new ArrayList<>();
+        skillList.add(skill);
+        job.setSkillList(skillList);
+        JobDto jobDto = jobMapper.toJobDto(job);
+        Optional<Job> jobsOptional = Optional.of(job);
+        List<Job> jobsList = new ArrayList<>();
+        jobsList.add(job);
+        List<JobDto> jobsDtoList = jobMapper.toJobDtoList(jobsList);
+        when(jobService.getAllJobs()).thenReturn(jobsList);
+        mockMvc.perform(MockMvcRequestBuilders.get("/jobs").
+                        contentType(MediaType.APPLICATION_JSON).
+                        content(asJsonString(jobsDtoList))).
+                andDo(MockMvcResultHandlers.print());
+        verify(jobService).getAllJobs();
+        verify(jobService,times(1)).getAllJobs();
+
+        when(jobService.getJobsByLocation("Koch")).thenReturn(jobsList);
+        mockMvc.perform(MockMvcRequestBuilders.get("/jobs?location=Koch").
+                        contentType(MediaType.APPLICATION_JSON).
+                        content(asJsonString(jobsDtoList))).
+                andDo(MockMvcResultHandlers.print());
+        verify(jobService).getJobsByLocation("Koch");
+        verify(jobService,times(1)).getJobsByLocation("Koch");
+
+        String[] skillSearch = new String[]{"singing"};
+        when(helper.getJobListFilteredBySkills(jobsList,skillSearch)).thenReturn(jobsList);
+        mockMvc.perform(MockMvcRequestBuilders.get("/jobs?skills=singing").
+                        contentType(MediaType.APPLICATION_JSON).
+                        content(asJsonString(jobsDtoList))).
+                andDo(MockMvcResultHandlers.print());
+        verify(helper).getJobListFilteredBySkills(jobsList,skillSearch);
+        verify(helper,times(1)).getJobListFilteredBySkills(jobsList,skillSearch);
+
+        when(helper.getJobListFilteredBySkills(jobsList,skillSearch)).thenReturn(jobsList);
+        mockMvc.perform(MockMvcRequestBuilders.get("/jobs?location=Koch&skills=singing").
+                        contentType(MediaType.APPLICATION_JSON).
+                        content(asJsonString(jobsDtoList))).
+                andDo(MockMvcResultHandlers.print());
+        verify(helper,times(2)).getJobListFilteredBySkills(jobsList,skillSearch);
+    }
```

Comment:
can you split it into different tests, ideally one test should handle only one scenario.

Reply:
okay sure